### PR TITLE
Reduce repo indexer disk usage

### PR DESCRIPTION
**Pull request by ethantkoenig**
Reduces disk usage of the repo (i.e. code) indexer:
- Disables bleve's `_all` field (which meant that we were previously storing everything twice)
- Use the bleve unique token filter (https://github.com/blevesearch/bleve/pull/739), since we only display the first occurrence of the search term.

I saw as roughly 3x (1.5GB -> 500MB) reduction in disk usage as a result of these changes (of course, mileage will vary depending on what type of text/code you are indexing).

Also introduces a migration-like versions to the issue and repo indexers to facilitate changes (which will typically require rebuilding the index).

Yes, this PR shamelessly pulls in https://github.com/ethantkoenig/rupture as a dependency to facilitate tracking indexer versions and migrations; I am aware of no other alternatives.

### CODE REVIEW

File: modules/indexer/issue.go [hunk 1]
```diff
@@ -70,9 +73,15 @@ func createIssueIndexer() error {
 	mapping := bleve.NewIndexMapping()
 	docMapping := bleve.NewDocumentMapping()
 
+	numericFieldMapping := bleve.NewNumericFieldMapping()
+	numericFieldMapping.Store = false
+	numericFieldMapping.IncludeInAll = false
 	docMapping.AddFieldMappingsAt("RepoID", bleve.NewNumericFieldMapping())
```
Comment (lafriks) [inline modules/indexer/issue.go]: Should this use `numericFieldMapping`?

Comment (ethantkoenig) [reply modules/indexer/issue.go]: Yes, fixed

**Review by lafriks — APPROVED** [review]

**Review by appleboy — CHANGES_REQUESTED** [review]

**Comment by ethantkoenig** [comment]
@appleboy Done

**Review by appleboy — APPROVED** [review]

**Comment by lafriks** [comment]
@ethantkoenig please resolve conflicts

**Comment by ethantkoenig** [comment]
@lafriks Resolved

**Pull request merged**